talestation_modules/code/modules/antagonists/bloodsuckers/antagonists/vassal/vassal_datum.dm

```diff
@@ -0,0 +1,316 @@
+	return "Master : [master.owner.name]"
+
+/datum/antagonist/vassal/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current_mob = mob_override || owner.current
+	current_mob.apply_status_effect(/datum/status_effect/agent_pinpointer/vassal_edition)
+	add_team_hud(current_mob)
+
+/datum/antagonist/vassal/add_team_hud(mob/target)
+	QDEL_NULL(team_hud_ref)
+
+	team_hud_ref = WEAKREF(target.add_alt_appearance(
+		/datum/atom_hud/alternate_appearance/basic/has_antagonist,
+		"antag_team_hud_[REF(src)]",
+		image(hud_icon, target, antag_hud_name),
+	))
+
+	var/datum/atom_hud/alternate_appearance/basic/has_antagonist/hud = team_hud_ref.resolve()
+
+	var/list/mob/living/mob_list = list()
+	mob_list += master.owner.current
+	for(var/datum/antagonist/vassal/vassal as anything in master.vassals)
+		mob_list += vassal.owner.current
+
+	for (var/datum/atom_hud/alternate_appearance/basic/has_antagonist/antag_hud as anything in GLOB.has_antagonist_huds)
+		if(!(antag_hud.target in mob_list))
+			continue
+		antag_hud.show_to(target)
+		hud.show_to(antag_hud.target)
+
+/datum/antagonist/vassal/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current_mob = mob_override || owner.current
+	current_mob.remove_status_effect(/datum/status_effect/agent_pinpointer/vassal_edition)
+
+/datum/antagonist/vassal/pre_mindshield(mob/implanter, mob/living/mob_override)
+	return COMPONENT_MINDSHIELD_PASSED
+
+/// This is called when the antagonist is successfully mindshielded.
+/datum/antagonist/vassal/on_mindshield(mob/implanter, mob/living/mob_override)
+	owner.remove_antag_datum(/datum/antagonist/vassal)
+	owner.current.log_message("has been deconverted from Vassalization by [implanter]!", LOG_ATTACK, color="#960000")
+	return COMPONENT_MINDSHIELD_DECONVERTED
+
+/datum/antagonist/vassal/proc/on_examine(datum/source, mob/examiner, examine_text)
+	SIGNAL_HANDLER
+
+	if(!iscarbon(source))
+		return
+	var/mob/living/carbon/carbon_source = source
+	var/vassal_examine = carbon_source.ReturnVassalExamine(examiner)
+	if(vassal_examine)
+		examine_text += vassal_examine
+
+/datum/antagonist/vassal/on_gain()
+	RegisterSignal(owner.current, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+	/// Enslave them to their Master
+	if(!master || !istype(master, master))
+		return
+	if(special_type)
+		if(!master.special_vassals[special_type])
+			master.special_vassals[special_type] = list()
+		master.special_vassals[special_type] |= src
+	master.vassals |= src
+	owner.enslave_mind_to_creator(master.owner.current)
+	owner.current.log_message("has been vassalized by [master.owner.current]!", LOG_ATTACK, color="#960000")
+	/// Give Recuperate Power
+	BuyPower(new /datum/action/bloodsucker/recuperate)
+	/// Give Objectives
+	var/datum/objective/bloodsucker/vassal/vassal_objective = new
+	vassal_objective.owner = owner
+	objectives += vassal_objective
+	/// Give Vampire Language & Hud
+	owner.current.grant_all_languages(FALSE, FALSE, TRUE)
```

Comment:
That just gives them omnitongue, so they can speak any language they already know. It's the alternative solution to a TG edit because tongues use a static list.